pkg/datapath/linux/probes/probes.go

```diff
@@ -201,85 +208,101 @@ func (p *ProbeManager) SystemKernelHz() (int, error) {
-	kernelParams["CONFIG_NET_SCH_INGRESS"] = config.ConfigNetSchIngress.Enabled()
-	kernelParams["CONFIG_NET_CLS_BPF"] = config.ConfigNetClsBpf.Enabled()
-	kernelParams["CONFIG_NET_CLS_ACT"] = config.ConfigNetClsAct.Enabled()
-	kernelParams["CONFIG_BPF_JIT"] = config.ConfigBpfJit.Enabled()
-	kernelParams["CONFIG_HAVE_EBPF_JIT"] = config.ConfigHaveEbpfJit.Enabled()
+	kernelParams["CONFIG_BPF"] = kernelOption{
+		Enabled:     config.ConfigBpf.Enabled(),
+		Description: coreInfraDescription,
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_BPF_SYSCALL"] = kernelOption{
+		Enabled:     config.ConfigBpfSyscall.Enabled(),
+		Description: coreInfraDescription,
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_NET_SCH_INGRESS"] = kernelOption{
+		Enabled:     config.ConfigNetSchIngress.Enabled() || config.ConfigNetSchIngress.Module(),
+		Description: coreInfraDescription,
+		CanBeModule: true,
+	}
+	kernelParams["CONFIG_NET_CLS_BPF"] = kernelOption{
+		Enabled:     config.ConfigNetClsBpf.Enabled() || config.ConfigNetClsBpf.Module(),
+		Description: coreInfraDescription,
+		CanBeModule: true,
+	}
+	kernelParams["CONFIG_NET_CLS_ACT"] = kernelOption{
+		Enabled:     config.ConfigNetClsAct.Enabled(),
+		Description: coreInfraDescription,
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_BPF_JIT"] = kernelOption{
+		Enabled:     config.ConfigBpfJit.Enabled(),
+		Description: coreInfraDescription,
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_HAVE_EBPF_JIT"] = kernelOption{
+		Enabled:     config.ConfigHaveEbpfJit.Enabled(),
+		Description: coreInfraDescription,
+		CanBeModule: false,
+	}
 
 	return kernelParams
 }
 
 // GetOptionalConfig performs a check of *optional* kernel configuration options. It
 // returns a map indicating which optional/non-mandatory kernel paramaters are enabled.
 // GetOptionalConfig is being used by CLI "cilium kernel-check".
-func (p *ProbeManager) GetOptionalConfig() map[KernelParam]bool {
+func (p *ProbeManager) GetOptionalConfig() map[KernelParam]kernelOption {
 	config := p.features.SystemConfig
-	kernelParams := make(map[KernelParam]bool)
+	kernelParams := make(map[KernelParam]kernelOption)
 
-	kernelParams["CONFIG_CGROUP_BPF"] = config.ConfigCgroupBpf.Enabled()
-	kernelParams["CONFIG_LWTUNNEL_BPF"] = config.ConfigLwtunnelBpf.Enabled()
-	kernelParams["CONFIG_BPF_EVENTS"] = config.ConfigBpfEvents.Enabled()
+	kernelParams["CONFIG_CGROUP_BPF"] = kernelOption{
+		Enabled:     config.ConfigCgroupBpf.Enabled(),
+		Description: "Host Reachable Services and Sockmap optimization",
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_LWTUNNEL_BPF"] = kernelOption{
+		Enabled:     config.ConfigLwtunnelBpf.Enabled(),
+		Description: "Lightweight Tunnel hook for IP-in-IP encapsulation",
+		CanBeModule: false,
+	}
+	kernelParams["CONFIG_BPF_EVENTS"] = kernelOption{
+		Enabled:     config.ConfigBpfEvents.Enabled(),
+		Description: "Visibility and introspection into datapath",
```

Comment:
Suggestion:
```suggestion
		Description: "Visibility and congestion management with datapath",
```
I think the "introspection" is kinda implied by the visibility. But `pkg/maps/signalmap` also handles some GC behaviour in some scenarios which is a bit more important from a datapath operations perspective. Making this a bit more visible might happen to help here if someone runs the tool and finds that CONFIG_BPF_EVENTS is disabled.

Reply:
@joestringer thanks for your inpot, changed accordingly.